pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -539,22 +539,22 @@ func xfrmKeyEqual(s1, s2 *netlink.XfrmState) bool {
 		bytes.Equal(s1.Auth.Key, s2.Auth.Key)
 }
 
-func ipSecReplaceStateIn(log *slog.Logger, localIP, remoteIP net.IP, nodeID uint16, zeroMark bool, localBootID, remoteBootID string, remoteRebooted bool, reqID int) (uint8, error) {
-	key := getNodeIPsecKey(localIP, remoteIP, localBootID, remoteBootID, netlink.XFRM_DIR_IN)
+func ipSecReplaceStateIn(log *slog.Logger, params *IPSecParameters) (uint8, error) {
+	key := getNodeIPsecKey(*params.SourceTunnelIP, *params.DestTunnelIP, params.LocalBootID, params.RemoteBootID, netlink.XFRM_DIR_IN)
```

Comment:
nit: this function looks suitable to become a method of *IPSecParameters

Reply:
I don't disagree, but this PR already makes a lot of changes. So may hold this one off until a follow-up.